plasma/root_chain/contracts/RootChain/RootChain.sol

```diff
@@ -157,84 +123,91 @@ contract RootChain {
     // @param sigs Both transaction signatures and confirmations signatures used to verify that the exiting transaction has been confirmed
     function startExit(uint256 utxoPos, bytes txBytes, bytes proof, bytes sigs)
         public
-        incrementOldBlocks
     {
         var txList = txBytes.toRLPItem().toList(11);
+        uint256 amount = txList[7 + 2 * oindex].toUint();
         uint256 blknum = utxoPos / 1000000000;
         uint256 txindex = (utxoPos % 1000000000) / 10000;
         uint256 oindex = utxoPos - blknum * 1000000000 - txindex * 10000;
+        address exitor = txList[6 + 2 * oindex].toAddress();
+        require(msg.sender == exitor);
         bytes32 root = childChain[blknum].root;
-
-        require(msg.sender == txList[6 + 2 * oindex].toAddress());
-        bytes32 txHash = keccak256(txBytes);
-        bytes32 merkleHash = keccak256(txHash, ByteUtils.slice(sigs, 0, 130));
-        require(Validate.checkSigs(txHash, root, txList[0].toUint(), txList[3].toUint(), sigs));
+        bytes32 merkleHash = keccak256(keccak256(txBytes), ByteUtils.slice(sigs, 0, 130));
+        require(Validate.checkSigs(keccak256(txBytes), root, txList[0].toUint(), txList[3].toUint(), sigs));
         require(merkleHash.checkMembership(txindex, root, proof));
+        addExitToQueue(utxoPos, exitor, amount);
+    }
 
-        // Priority is a given utxos position in the exit priority queue
+    // Priority is a given utxos position in the exit priority queue
+    function addExitToQueue(uint256 utxoPos, address exitor, uint256 amount)
+        private
+    {
+        uint256 blknum = utxoPos / 1000000000;
+        uint256 txindex = (utxoPos % 1000000000) / 10000;
+        uint256 oindex = utxoPos - blknum * 1000000000 - txindex * 10000;
         uint256 priority;
-        if (blknum < weekOldBlock) {
-            priority = (utxoPos / blknum).mul(weekOldBlock);
+        if (childChain[blknum].created_at - 1 weeks > block.timestamp) {
```

Comment:
I believe this check should be:

`if (block.timestamp - 1 weeks > childChain[blknum].created_at)`

I think it would be better to use @bun919tw's suggestion (`max(..)`)